pageserver/src/tenant/timeline/walreceiver/walreceiver_connection.rs

```diff
@@ -127,20 +122,24 @@ pub(super) async fn handle_walreceiver_connection(
         "walreceiver connection",
         false,
         async move {
+            debug_assert_current_span_has_tenant_and_timeline_id();
+
             select! {
                 connection_result = connection => match connection_result {
-                    Ok(()) => info!("Walreceiver db connection closed"),
+                    Ok(()) => debug!("Walreceiver db connection closed"),
                     Err(connection_error) => {
-                        if let Err(e) = ignore_expected_errors(connection_error) {
-                            warn!("Connection aborted: {e:#}")
+                        if connection_error.is_expected() {
```

Comment:
if !connection_error.is_expected() {
  warn!("Connection aborted: {connection_error:#}")
}

Reply:
I added the true branch first because this is a peculiar place, compared to others which always log an error (info or error). I don't know why it is silenced, but previously it clearly was. I agree that doing your version cuts the lines of code down, but I wanted to keep the intentional silencing.